src/hotspot/share/prims/upcallLinker.cpp

```diff
@@ -138,10 +136,9 @@ void UpcallLinker::on_exit(UpcallStub::FrameData* context) {
 }
 
 void UpcallLinker::handle_uncaught_exception(oop exception) {
-  ResourceMark rm;
-  // Based on CATCH macro
   tty->print_cr("Uncaught exception:");
-  exception->print();
+  Handle exception_h(Thread::current(), exception);
+  java_lang_Throwable::print_stack_trace(exception_h, tty);
   ShouldNotReachHere();
```

Comment:
How does `print_stack_trace` not return here?

Reply:
It does return. `ShouldNotReachHere` is used to crash the VM.

Reply:
`fatal()` might be better here. I could change it.

Reply:
Yes please do.